include/vk_mem_alloc.h

```diff
@@ -17570,8 +17569,8 @@ struct AtomicTransactionalIncrement
 VkResult VmaAllocator_T::AllocateVulkanMemory(const VkMemoryAllocateInfo* pAllocateInfo, VkDeviceMemory* pMemory)
 {
     AtomicTransactionalIncrement<uint32_t> deviceMemoryCountIncrement;
-    const uint64_t prevDeviceMemoryCount = deviceMemoryCountIncrement.Increment(&m_DeviceMemoryCount);
 #if VMA_DEBUG_DONT_EXCEED_MAX_MEMORY_ALLOCATION_COUNT
+    const uint64_t prevDeviceMemoryCount = deviceMemoryCountIncrement.Increment(&m_DeviceMemoryCount);

```

Comment:
This is wrong, it will now skip the increment if the debug option is not on.

Reply:
Cases like this are exactly why declare I am not going to be free of any compiler warnings. Also not going to make the code more complicated (like capturing the returned value to a variable only #if...). I prefer to just silence such warnings.

Reply:
You could add a `VMA_DEBUG_ASSERTED` macro that expands to `__attribute__((unused))` when debugging is disabled so the compiler will shut up but if you remove the debug use will then complain.

Reply:
This is what I mean by complicating the code just to silence compiler warnings. I'm sorry but this is explicitly defined as out-of-scope of this library in the documentation.